crates/sui-types/src/indirect_validity.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::committee::EpochId;
+use crate::messages_checkpoint::CheckpointSequenceNumber;
+
+use serde::{Deserialize, Serialize};
+
+/// IndirectValidity is a placeholder for signatures, which indicates that the wrapped message has
+/// been proven valid through indirect means, typically inclusion in a certified checkpoint or
+/// via f+1 votes that the message is correct.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct IndirectValidity(Validity);
```

Comment:
Depending on how you plan to use these types, consider name them as:
IndirectValidity -> CertificateProof
Validity -> CertificateProofKind